### ISSUE TITLE
GALILEO HAS started 2023-01-24

### ISSUE THREAD

**Issue by bernieEV** [barbeau/gpstest#614]
Hi Sean,
I am in Europe/Austria and got GALILEO navigation info in the news.

please check this service started 2023-01-24
https://www.gsc-europa.eu/galileo/services/galileo-high-accuracy-service-has

Is it possible to add this to GPStest app ?
thanks

**Comment by bernieEV** [comment]
Hi Sean,
I checked my HW.
It is "Google PIXEL6" and your program "GPStest" tells in the HELP "Broadcom BCM4776"
looking into the Broadcom specs it can receive european GALILEO frequency E1 and E5a.
looking into the GALILEO frequency plan the new service HAS is in the frequency E6b.
unfortunately my PIXEL6 cannot receive E6b.
best regards
Bernie

**Comment by danieldjewell** [comment]
While @barbeau is definitely more of an expert than I am, my own experience with Android devices is that there is not *necessarily* a correlation between what a receiver chip might support (on paper) and what it will support, as configured by an OEM.

For example, my US-model Samsung S22 Ultra has a Snapdragon 8 Gen 1 ... which, according to the datasheet, supports "GPS, Glonass, BeiDou, Galileo, QZSS, NavIC" and "Dual Frequency GNSS (L1/L5)" ... When I run GPSTest, I see zero BeiDou, and only L1 Glonass. (Admittedly, it does say "L1/L5" and L2 Glonass is on a very different frequency from GPS/Galileo L1/L5...)

Without digging deep into the hardware configuration, some of the features might be disabled/inaccessible - even though the _chip_ might be capable...

**Comment by barbeau** [comment]
Sorry for the delayed response - @danieldjewell is right, there are a lot of reasons why the specific hardware specs might differ from the ultimate phone features. I actually wrote an article about this:
https://barbeau.medium.com/crowdsourcing-gnss-capabilities-of-android-devices-d4228645cf25

@bernieEV unfortunately yes, Galileo HAS is on E6b, which is a frequency phones don't support.

**Comment by jidanni** [comment]
@barbeau one moment please.                                                                                
                                                                                                     
I am all excited about the Galileo High Accuracy Service.                                            
I have a, yes,                                                                                       
- [compatible cellphone](https://www.usegalileo.eu/EN/inner.html#data=smartphone)                    
- Internet connection                                                                                
                                                                                                     
OK, then please document what I need to do to see                                                    
                                                                                                     
Longitude: 123.1234567                                                                           
Latitude: 88.1234567                                                                             
Accuracy: 20cm                                                                                   
                                                                                                     
on my screen. Thanks!

**Comment by barbeau** [comment]
tl;dr - HAS requires extra processing when computing positions from raw measurements and isn't available in GPSTest. Given the amount of work this takes I'm not likely to support it in GPSTest anytime soon. There has however been progress in terms of using HAS with smartphones - more details:

https://www.ion.org/gnss/abstracts.cfm?paperID=15966

Also, I commented above that HAS isn't available on smartphones because E6b isn't supported on smartphones. This is true that it cant receive HAS via GNSS signal broadcast, but it's also true that HAS is available over the internet as well, so as long as the phone is Internet connected it can receive HAS data.
